pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -3086,11 +3134,25 @@ _ReadPolyMesh(_PrimReaderContext* context)
         SdfValueTypeNames->IntArray,
         _CopyGeneric<IInt32ArrayProperty, int>(
             context->ExtractSchema(".faceCounts")));
-    context->AddProperty(
-        _GetUVPropertyName(),
-        _GetUVTypeName(),
-        _CopyGeneric<IV2fGeomParam, GfVec2f>(
-            context->ExtractSchema("uv")));
+
+    // special case for texture coordinates: we want to preserve the topology,
+    // so we need to extract the indices properties if they exist.
+    auto uvProperty = context->ExtractSchema("uv");
+    if (uvProperty.Cast<IV2fGeomParam>().isIndexed()) {
+        context->AddProperty(
+            _GetUVPropertyName(),
+            _GetUVTypeName(),
+            _CopyGeneric<IV2fGeomParam, GfVec2f, false>(uvProperty));
+        context->AddProperty(
+            TfToken(_GetUVPropertyName().GetString() + ":indices"),
```

Comment:
There's a UsdGeomTokens->indices, which you can use in combination with [SdfPath::JoinIdentifier)[https://github.com/PixarAnimationStudios/USD/blob/722956654a7090c23ac33d6e52ae93e05e38f94b/pxr/usd/lib/sdf/path.h#L597]